### ISSUE TITLE
`int_ranges`: PanicException and InvalidOperationError

### ISSUE THREAD

**Issue by JulianCologne** [pola-rs/polars#11891]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
(
    pl.DataFrame({
        'asc': [1, 2, 3],
        'desc': [3, 2, 1],
        'zero': [0, 0, 0],
        'one': [1, 1, 1],
        'two': [2, 2, 2],
    })
    .with_columns(
        zero_asc=pl.int_ranges(start='zero', end='asc'),
        zero_desc=pl.int_ranges(start='zero', end='desc'),

        asc_desc=pl.int_ranges(start='asc', end='desc'),
        desc_asc=pl.int_ranges(start='desc', end='asc'),

        asc_two=pl.int_ranges(start='asc', end='two'),

        # <<< InvalidOperationError: range must be decreasing if 'step' is negative >>>
        # asc_desc_down=pl.int_ranges(start='asc', end='desc', step=-1),
        # desc_asc_down=pl.int_ranges(start='desc', end='asc', step=-1),

        # <<< # PanicException: capacity overflow; [thread '<unnamed>' panicked at library/alloc/src/raw_vec.rs:534:5:] >>>
        # asc_zero_down=pl.int_ranges(start='asc', end='zero', step=-1),
        # asc_one=pl.int_ranges(start='asc', end='one'),
    )
)
┌─────┬──────┬──────┬─────┬───┬───────────┬───────────┬───────────┬───────────┐
│ asc ┆ desc ┆ zero ┆ one ┆ … ┆ zero_desc ┆ asc_desc  ┆ desc_asc  ┆ asc_two   │
│ --- ┆ ---  ┆ ---  ┆ --- ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---       │
│ i64 ┆ i64  ┆ i64  ┆ i64 ┆   ┆ list[i64] ┆ list[i64] ┆ list[i64] ┆ list[i64] │
╞═════╪══════╪══════╪═════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
│ 1   ┆ 3    ┆ 0    ┆ 1   ┆ … ┆ [0, 1, 2] ┆ [1, 2]    ┆ []        ┆ [1]       │
│ 2   ┆ 2    ┆ 0    ┆ 1   ┆ … ┆ [0, 1]    ┆ []        ┆ []        ┆ []        │
│ 3   ┆ 1    ┆ 0    ┆ 1   ┆ … ┆ [0]       ┆ []        ┆ [1, 2]    ┆ []        │
└─────┴──────┴──────┴─────┴───┴───────────┴───────────┴───────────┴───────────┘
```

### InvalidOperationError: range must be decreasing if 'step' is negative
if ranges `1->3 = [1, 2]` and `3->1 = []`
then with `step=-1` should be: `1->3 = []` and `3->1 = [3, 2]` and not error

### PanicException: capacity overflow
going from `1/2/3` down to zero with `step=-1` should work and not panic
`1->0 = [1]`, `2->0 = [2, 1]`, `3->0 = [3, 2, 1]`
going from `1/2/3` to `1`  should work and not panic 
`1->1 = []`, `2->1 = []`, `3->1 = []`

### Expected behavior

All examples should work without exception or panic

### Installed versions

<details>

```
Replace this line with the output of pl.show_versions(). Leave the backticks in place.
```

</details>

**Comment by mcrumiller** [comment]
Sorry, do you mean that we should return nulls in cases where the operation is invalid, and return value elsewhere?

**Comment by JulianCologne** [comment]
> Sorry, do you mean that we should return nulls in cases where the operation is invalid, and return value elsewhere?

no ;)

We discussed this here XXX and we all agreed that `<xxx>_ranges` should behave like python/rust ranges like this:
- no exceptions / no nulls (given that data types are correct)
- for ascending ranges:
  - `3-6 = [3, 4, 5]`
  - `6-3 = []` end is lower than start: no problem, there is no element in between so empty list
- for descending ranges
  - `3-6 = []` end is higher than start: no problem, there is no element in between so empty list
  - `3-6 = [6, 5, 4]`

**Comment by alexander-beedie** [comment]
Closed by #12548 👌
